esp-hal-common/src/uart.rs

```diff
@@ -1645,13 +1679,17 @@ mod asynch {
         /// # Ok
         /// When successful, returns the number of bytes written to buf.
         /// This method will never return Ok(0), unless buf.len() == 0.
-        async fn read_async(&mut self, buf: &mut [u8]) -> Result<usize, Error> {
+        pub async fn read_async(&mut self, buf: &mut [u8]) -> Result<usize, Error> {
             if buf.len() == 0 {
-                return Ok(0);
+                return Err(Error::InvalidArgument);
```

Comment:
What's the driving force behind this change? Imo `Ok(0)` is fine, but maybe I'm missing something.

Reply:
Ah it was during debugging, Was unsure if it got stuck in a infinite loop or not so had it report back invalid argument if an empty buffer was provided. Then I started considering if its good to leave it there as anyone who would use it would probably be having issues in their implementation if they are sending in a zero length buffer.

Reply:
I took another read of the [embedded-io docs](https://github.com/rust-embedded/embedded-hal/blob/5e21d566e25049a85a74010e0c92708f732b2a7f/embedded-io-async/src/lib.rs#L39-L40) just because most people are going to use this read method through that trait. It says `Ok(0)` or an error is fine. I think an error is better here because I can picture an scenario where one my enter a hot loop by trying to repeatedly read into an empty buffer.

So we'll keep it as an error.